File: sandbox-tests/code-bugfixing/password-validator/C#/PasswordValidatorTest.cs
using NUnit.Framework;


namespace Tests
{
    [TestFixture]
    public class PasswordValidatorTest
    {
        [Test]
        public void TestValidPasswords()
        {
            Assert.IsTrue(PasswordValidator.IsValidPassword("Password1!"));
            Assert.IsTrue(PasswordValidator.IsValidPassword("StrongPass123$"));
            Assert.IsTrue(PasswordValidator.IsValidPassword("Good#Pass1"));
            Assert.IsTrue(PasswordValidator.IsValidPassword("Valid1@Pass"));
        }

        [Test]
        public void TestInvalidPasswords()
        {
            Assert.IsFalse(PasswordValidator.IsValidPassword("Pass1!"));
            Assert.IsFalse(PasswordValidator.IsValidPassword("password1!"));
            Assert.IsFalse(PasswordValidator.IsValidPassword("PASSWORD1!"));
            Assert.IsFalse(PasswordValidator.IsValidPassword("Password!"));
            Assert.IsFalse(PasswordValidator.IsValidPassword("Password1"));
            Assert.IsFalse(PasswordValidator.IsValidPassword("Password 1!"));
            Assert.IsFalse(PasswordValidator.IsValidPassword(null));
        }
    }
}

File: sandbox-tests/code-bugfixing/password-validator/java/PasswordValidatorTest.java
package com.aicode.java;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class PasswordValidatorTest {

    @Test
    public void testValidPasswords() {
        assertTrue(PasswordValidator.isValidPassword("Password1!"));
        assertTrue(PasswordValidator.isValidPassword("StrongPass123$"));
        assertTrue(PasswordValidator.isValidPassword("Good#Pass1"));
        assertTrue(PasswordValidator.isValidPassword("Valid1@Pass"));
    }

    @Test
    public void testInvalidPasswords() {
        assertFalse(PasswordValidator.isValidPassword("Pass1!"));
        assertFalse(PasswordValidator.isValidPassword("password1!"));
        assertFalse(PasswordValidator.isValidPassword("PASSWORD1!"));
        assertFalse(PasswordValidator.isValidPassword("Password!"));
        assertFalse(PasswordValidator.isValidPassword("Password1"));
        assertFalse(PasswordValidator.isValidPassword("Password 1!"));
        assertFalse(PasswordValidator.isValidPassword(null));
    }
}